File: Python/fibonacci.py
def fib(n, basCase):
    '''
    Given an n [int] and base case [dict] 
    contraint: n >= 1
    @return the nth fibonacci number
    '''
    if n in basCase.keys():
        return basCase[n]
    else:
        basCase[n] = fib(n - 1, baseCase) + fib(n - 2, baseCase)
        return basCase[n]
        
baseCase = {
    1 : 1,
    2 : 1
}

print(fib(6, baseCase))

File: Golang/Fibonacci.go
package main

import "fmt"

// Fibonacci implementation in Golang
// TC - O(2*N) | SC - O(1)
func fib(val int) int {
	if val == 0 {
		return 0
	}
	if val < 0 {
		return 0
	}
	if val == 1 {
		return 1
	}
	return fib(val-2) + fib(val-1)
}

func main() {
	fmt.Println(fib(12))
	fmt.Println(fib(2))
	fmt.Println(fib(18))
	fmt.Println(fib(-21))
}
